File: UI/MyInterviewPreparation/ClientApp/src/Entities/UserDomain.ts
import { DomainStatus } from "./Enums";

export interface UserDomain {
    id: number;
    createdDate: string | null;
    updatedDate: string | null;
    name: string;
    domainStatus: DomainStatus;
    subjectId: number | null;
    systemDomainId: number | null;
    questionAnswerCount: number;
}

File: Common/CommonEntities/QuestionAnswer/UserDomain.cs
using CommonEntities.Base;
using static CommonEntities.Enums;

namespace CommonEntities.QuestionAnswer
{
    public class UserDomain : BaseRequest
    {
        public int Id { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public string Name { get; set; }
        public DomainStatus DomainStatus { get; set; }
        public int? SubjectId { get; set; }
        public int? SystemDomainId { get; set; }
        public int QuestionAnswerCount { get; set; }
    }
}
